### ISSUE TITLE
Start video playback at arbitrary position

### ISSUE THREAD

**Issue by theychx** [ur1katz/casttube#5]
Hi
The default media receiver supports starting playback at an arbitrary position in content. I assume that the Youtube cc app also does this. Support for this in ```casttube``` would be useful.

**Comment by ur1katz** [comment]
I just opened a small branch that starts a video at a specified time. I can also add the option to start a playlist at a specific index if you need it.
i'm planning on adding some more features(closed caption support, autoplay toggles and maybe the playlist data from #4 ) in the next few weeks so i won't create a release until then. I can merge this branch before that if it helps :)

**Comment by theychx** [comment]
Amazing, thanks!
(it makes no difference to me whether the branch is merged or not)

**Comment by ur1katz** [comment]
@theychx I know it's been long but I have an open PR. Iv'e tested this locally but could you please help me verify this is working on other setups as well?

**Comment by theychx** [comment]
I've implemented start time in [this fork](https://github.com/theychx/catt).
It works, but I sometimes get a strange issue with the start time of the previous video being added to the start time of a new video. For example when alternating between these commands:
```
catt cast -t 30 http://www.dailymotion.com/video/x6fotne
catt cast -t 45:0 https://www.youtube.com/watch?v=hiAkdoETqNA
```
(and yes, I did check whether my code passes the correct value :smiley: )

I'll push more code to the fork later today, for testing your queue data PR.

**Comment by theychx** [comment]
I'm using [this modified controller](https://gist.github.com/theychx/ac628691375f2dc30937c11dcb5222cc).

**Comment by ur1katz** [comment]
Thanks!, the fork looks really good.
I ran these commands(Same as your comment but the YouTube video is a timer) in a loop
```
catt cast -t 30 http://www.dailymotion.com/video/x6fotne
catt cast -t 45:0 https://www.youtube.com/watch?v=2uMc3rNnTo4
``` 
I couldn't recreate the problem you mentioned. the timer was always 15. Did you do anything other than just alternating them?

**Comment by theychx** [comment]
As it happens, I can only reproduce this with my cc ultra, and not with my regular 2nd gen cc. Which is weird and really annoying. So there's a problem somewhere (perhaps in the youtube cc-app itself), that only affects ultra users.

**Comment by ur1katz** [comment]
That's weird! I can't test this on my setup(maybe it's a good excuse to upgrade :D) If you have the time could you check a few things?
1. Does it happen every time?
2. Does it reproduce with the YouTube android app?(seek a video then cast it)
3. Is the time always the same? if so at what time does the video start?

Many thanks!

**Comment by theychx** [comment]
> Does it happen every time?

It only happens when the previous content was played by the ```Default Media Receiver```.
>  Does it reproduce with the YouTube android app?

Nope (regardless of previous content).
> Is the time always the same? if so at what time does the video start?

Now this is super weird. It seems that for every iteration of playing content with the default cc-app and the youtube ditto, the starting time is pushed an additional ~30 secs. I've tried different starting times for the non-youtube content, but that doesn't seem to affect the problem.

Just to be clear: Arbitrary starting times always work with the ```Default Media Receiver```.

**Comment by ur1katz** [comment]
The problem we had was a bug in the YouTube app for chromecast on some devices. From all of my tests they fixed the bug so iv'e included the feature in release 0.2.1
